de-lib/src/main/java/org/iplantc/de/apps/client/events/tools/ToolSearchResultLoadEvent.java

```diff
@@ -0,0 +1,64 @@
+package org.iplantc.de.apps.client.events.tools;
+
+import org.iplantc.de.client.models.tool.Tool;
+import com.google.common.base.Preconditions;
+import com.google.gwt.event.shared.EventHandler;
+import com.google.gwt.event.shared.GwtEvent;
+import com.google.gwt.event.shared.HandlerRegistration;
+
+import java.util.List;
+
+/**
+ * A GwtEvent used to notify listeners that Tool search results have been loaded from the search service.
+ * 
+ * @author sriram
+ * 
+ */
+public class ToolSearchResultLoadEvent extends GwtEvent<ToolSearchResultLoadEvent.ToolSearchResultLoadEventHandler> {
+
+    public interface ToolSearchResultLoadEventHandler extends EventHandler {
+
+        void onToolSearchResultLoad(ToolSearchResultLoadEvent event);
+    }
+
+    public static interface HasToolSearchResultLoadEventHandlers {
+        HandlerRegistration addToolSearchResultLoadEventHandler(ToolSearchResultLoadEventHandler handler);
+    }
+
+
+    public static final Type<ToolSearchResultLoadEventHandler> TYPE = new Type<>();
+    private final String searchPattern;
+    private final List<Tool> results;
+    private final String searchText;
+
+    public ToolSearchResultLoadEvent(final String searchText,
+                                    final String searchPattern,
+                                    final List<Tool> results) {
+        Preconditions.checkNotNull(results);
```

Comment:
I like the use of the Preconditions class 👍   If that condition fails though, it will throw an error which may or may not be what you want for some of the use cases in this PR.  Just throwing that out there.